src/openzaak/components/autorisaties/forms.py

```diff
@@ -322,12 +322,13 @@ def _validate_external_types(self, component):
                 )
                 self.add_error("externe_typen", error)
 
-    def get_types(self):
-        component = self.cleaned_data["component"]
+    def get_types(self, component):
         related_type_selection = self.cleaned_data.get("related_type_selection")
         types = None
         if related_type_selection:
-            _field_info = COMPONENT_TO_FIELDS_MAP[component]
+            _field_info = COMPONENT_TO_FIELDS_MAP.get(component)
+            if _field_info is None:
```

Comment:
I'm not sure if this check is required. If `related_type_selection` is filled, that means that some types should be chosen. And if not, it's an error.

Reply:
i agree, will fix